riscv/triggers.h

```diff
@@ -83,6 +83,7 @@ class trigger_t {
   virtual std::optional<match_result_t> detect_memory_access_match(processor_t UNUSED * const proc,
       operation_t UNUSED operation, reg_t UNUSED address, std::optional<reg_t> UNUSED data) noexcept { return std::nullopt; }
   virtual std::optional<match_result_t> detect_trap_match(processor_t UNUSED * const proc, const trap_t UNUSED & t) noexcept { return std::nullopt; }
+  bool mode_match(state_t * const state) const noexcept;
```

Comment:
This can be `protected`, yes?

Reply:
Should we move the mode_match() checking to module_t along with textra_match()?
```auto result = trigger->textra_match(proc) ? trigger->detect_trap_match(proc, t) : std::nullopt;```
->
```auto result = (trigger->mode_match(proc->get_state()) && trigger->textra_match(proc)) ? trigger->detect_trap_match(proc, t) : std::nullopt;```

Reply:
You could even put it into `textra_match()` (and rename it to `mode_and_textra_match()`) since it's needed in both places that `textra_match()` is called.

But let's do that as a new PR.

Reply:
OK will do

Reply:
After adding trap_common_t, I think move textra_match to detect_??_match() functions makes more sense.
(https://github.com/riscv-software-src/riscv-isa-sim/pull/1207)